libs/wingsdk/src/target-tf-gcp/function.ts

```diff
@@ -95,6 +113,25 @@ export class Function extends cloud.Function {
       timeout: props.timeout?.seconds ?? 60,
       environmentVariables: props.env ?? {},
     });
+
+    // apply permissions to bound resources
+    if (this.permissions) {
+      for (const [uniqueId, roleDefinitions] of this.permissions) {
+        const scopedResource = this.node.tryFindChild(uniqueId);
```

Comment:
I'm wondering if I can find the resource this way. Need suggestions!!

Reply:
@tsuf239 any tips here?

Reply:
@Chriscbr?

Reply:
@subh-cs can you share more details about what you are trying to implement? It looks like some of the code is duplicated with the logic in the `addPermission` method